webui/src/pages/admin-dashboard/extension-version-container.tsx

```diff
@@ -0,0 +1,128 @@
+    const [allChecked, setAllChecked] = useState(false);
+
+    useEffect(() => {
+        setVersions(getVersions());
+    }, [props.extension]);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newVersionMap = new Map<string, boolean>();
+        versions.forEach((checked, version) => {
+            if (version === event.target.name) {
+                checked = event.target.checked;
+                if (allChecked && !checked) {
+                    setAllChecked(false);
+                }
+            } else if (event.target.name === 'checkAll') {
+                checked = event.target.checked;
+                setAllChecked(event.target.checked);
+            }
+            newVersionMap.set(version, checked);
+        });
+        setVersions(newVersionMap);
+    };
+
+    const handleUpdate = () => {
+        props.onUpdate();
+    };
+
+    return <>
+        <Grid container direction='column'>
+            <Grid item container>
+                {
+                    extension.files.icon ?
+                        <Grid item xs={12} md={4}>
+                            <img src={extension.files.icon}
+                                className={classes.extensionLogo}
+                                alt={extension.displayName || extension.name} />
+                        </Grid>
+                        : ''
+                }
+                <Grid item container xs={12} md={8}>
+                    <Grid item container direction='column' justify='center'>
+                        <Grid item>
+                            <Typography variant='h5' className={classes.titleRow}>
+                                {extension.displayName || extension.name}
+                            </Typography>
+                        </Grid>
+                        <Grid item>
+                            <Typography className={classes.code}>{extension.namespace}.{extension.name}</Typography>
+                        </Grid>
+                        <Grid item>
+                            <Typography classes={{ root: classes.description }}>{extension.description}</Typography>
+                        </Grid>
+                    </Grid>
+                </Grid>
+            </Grid>
+            <Grid item container>
+                <Grid item xs={12} md={4}></Grid>
+                <Grid item container xs={12} md={8} direction='column'>
+                    <FormControl component='fieldset'>
+                        <FormGroup>
+                            <FormControlLabel
+                                control={<Checkbox checked={allChecked} onChange={handleChange} name='checkAll' />}
+                                label='All Versions'
+                            />
+                            {
+                                Array.from(versions.entries()).filter(([version, checked]) => VERSION_ALIASES.indexOf(version) < 0).map(([version, checked], key) => {
```

Comment:
The versions should not only be filtered here, but everywhere in this component. For example, when I select "All Versions", the confirmation dialog still shows `latest` and `preview`.